src/invidious/playlists.cr

```diff
@@ -437,17 +437,23 @@ def get_playlist_videos(db, playlist, offset, locale = nil, continuation = nil)
     db.query_all("SELECT * FROM playlist_videos WHERE plid = $1 ORDER BY array_position($2, index) LIMIT 100 OFFSET $3",
       playlist.id, playlist.index, offset, as: PlaylistVideo)
   else
-    if offset >= 100
-      # Normalize offset to match youtube's behavior (100 videos chunck per request)
-      offset = (offset / 100).to_i64 * 100_i64
+    if continuation
+      initial_data = request_youtube_api_next(continuation, playlist.id)
+      offset = initial_data["contents"]["twoColumnWatchNextResults"]["playlist"]["playlist"]["currentIndex"].as_i
```

Comment:
Please switch to the [`#dig()` method of JSON::Any](https://crystal-lang.org/api/1.0.0/JSON/Any.html#dig(index_or_key:String%7CInt,*subkeys):JSON::Any-instance-method). See #2231